hypervisor/arch/x86/pagetable.c

```diff
@@ -249,7 +249,7 @@ static void add_pte(const uint64_t *pde, uint64_t paddr_start, uint64_t vaddr_st
 		uint64_t *pte = pt_page + index;
 
 		if (mem_ops->pgentry_present(*pte) != 0UL) {
-			ASSERT(false, "invalid op, pte present");
+			pr_fatal("%s, pte 0x%llx is already present!\n", __func__, vaddr);
```

Comment:
fatal ? You said this may happen normally

Reply:
No. I didn't say that. I said "it may happen (if the guest is not trusted and it will re-program the PCI BAR arbitrarily)".